src/main/java/com/google/devtools/build/lib/rules/cpp/CcLinkingHelper.java

```diff
@@ -312,6 +313,19 @@ public CcLinkingHelper setLinkedArtifactNameSuffix(String suffix) {
     return this;
   }
 
+  /*
+   * Adds a suffix (_{hash}) for path of linked DLL. This will be enabled only when rename_dll is enabled
+   * and the linked artifact's LinkTargetType is NODEPS_DYNAMIC_LIBRARY (i.e DLL generated by cc_library).
+   * We have to add the suffix to make sure the CppLinkAction link against the renamed DLL. If not, CppLinkAction
+   * will link against the DLL whose name is the same as the name of cc_library.
+   */
+  public CcLinkingHelper setLinkedDLLNameSuffix(String suffix) {
+    if(this.cppConfiguration.isRenameDLL(featureConfiguration)) {
```

Comment:
Why do we need to check it again after we already check if there should be a suffix in CppHelper?